src/js/line.js

```diff
@@ -0,0 +1,521 @@
+        });
+
+        // Exit any removed circle groups
+        var removedPointGroups = that.chartLinePointGroups.exit();
+        that.exitD3Elements(removedPointGroups);
+
+        // Append needed circles for each group
+        that.chartLinePointGroups.each(function (d, idx) {
+            var currentGroup = d3.select(this);
+
+            var circles = currentGroup.selectAll("circle")
+                    .data(d.data, function (currentData, index) {
+                    return d.id + "-" + index;
+                });
+
+            // Create new circles
+            circles.enter()
+            .append("circle")
+            .attr({
+                "class": chartLinePointClass,
+                "fill": color(idx),
+                "r": pointRadius,
+                "cy": function (d) {
+                    return yScale(d.value);
+                },
+                "cx": function (d) {
+                    return xScale(new Date(d.date));
+                }
+            })
+            .each(function (currentData, index) {
+                that.trackD3BoundElement(d.id + "-" + index, this);
+            });
+
+            // Remove exited circles
+            var removedCircles = circles.exit();
+            that.exitD3Elements(removedCircles);
+
+        });
+
+
+
+        // Transition circles
+        that.chartLinePointGroups.each(function () {
+            var currentGroup = d3.select(this);
+
+            currentGroup.selectAll("circle")
+                .transition().duration(transitionLength)
+                .attr({
+                    "cy": function (d) {
+                        return yScale(d.value);
+                    },
+                    "cx": function (d) {
+                        return xScale(new Date(d.date));
+                    }
+                });
+        });
+
+    };
+
+    floe.chartAuthoring.lineChart.chart.getYScale = function (that) {
+        var height = that.options.svgOptions.height;
+        var padding = that.options.lineOptions.padding;
+        var dataSet = that.model.wrappedDataSet;
+
+        var maxValues = [];
+
+        fluid.each(dataSet, function (setItem) {
+            var maxValue = d3.max(setItem.data, function (d) {
+                return d.value;
+            });
+            maxValues.push(maxValue);
+        });
+
+        var dataSetMax = d3.max(maxValues, function (d) {
+            return d;
+        });
+
+        return d3.scale.linear()
+            .domain([0, dataSetMax])
+            .nice()
+            .range([height - padding, padding]);
+    };
+
+    floe.chartAuthoring.lineChart.chart.getXScale = function (that) {
+        var width = that.options.svgOptions.width;
+        var padding = that.options.lineOptions.padding;
+        var dataSet = that.model.wrappedDataSet;
+
+        var minDates = [];
+        var maxDates = [];
+
+        fluid.each(dataSet, function (setItem) {
+            var minDate = d3.min(setItem.data, function (d) {
```

Comment:
I don't believe `Math.max/Math.min` handle arrays like the D3 min/max, based on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max

I agree on this being repetitive though - I'll see if I can figure out a better approach to the task (determining the max/min for range setting from a group of arrays).